style-guide.md

```diff
@@ -0,0 +1,165 @@
+# edX JavaScript Conventions
+
+# TODO
+
+I'm looking for widespread agreement on everything contained in this document!
+In particular, I'd love feedback on the [Goals](#goals), [Style](#style), and
+[Conventions](#conventions) sections.
+
+Here are a few things that would be awesome to have:
+- [ ] define and link to a jshint/jscs/eslint config
+- [ ] clarify path towards ES2015
+
+## Motivation
+
+Code conventions are important.  It helps us intuitively approach new code,
+revisit old code, and share code with others.
+
+## Goals
+
+We hope to write JavaScript which is:
+
+1. Legible
+2. Friendly
+3. Predictable
+4. Semantic
+
+## About
+
+This guide is split up into two sections - [Style](#style) and
+[Conventions](#conventions).  The style section addresses aesthetic topics,
+while the conventions section addresses topics that have more semantic
+significance.
+
+## Note: ECMAScript Version
+
+Currently, edX only supports ES5.  In order to implement modules, we use AMD by
```

Comment:
The discussion of the kind of modules we support is somewhat separate from our Javascript syntax support (ES5/ES2015) - we will probably support writing ES2015 soon, but will not change the way we do modules to CommonJS or ES6 modules for quite some time.